pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -536,14 +536,37 @@ func (r *ReconcileTailoredProfile) getRulesFromSelections(tp *cmpv1alpha1.Tailor
 	}
 	return rules, nil
 }
-
 func (r *ReconcileTailoredProfile) getVariablesFromSelections(tp *cmpv1alpha1.TailoredProfile, pb *cmpv1alpha1.ProfileBundle) ([]*cmpv1alpha1.Variable, error) {
-	variableList := []*cmpv1alpha1.Variable{}
-	for _, setValues := range tp.Spec.SetValues {
+	// 1) First pass over tp.Spec.SetValues to detect duplicates, warn, and decide which value to keep
+	//    We choose to keep the last occurrence. The map "duplicates" maps variableName -> finalValue.
+	//    The slice "uniqueVarOrder" tracks the first time we saw the variable (for stable iteration).
+	duplicates := make(map[string]string)
+	uniqueVarOrder := []string{}
+
+	for i, setValue := range tp.Spec.SetValues {
+		if oldVal, seen := duplicates[setValue.Name]; seen {
+			// We found a duplicate. Warn that we will ignore the previous usage.
+			r.Eventf(tp, corev1.EventTypeWarning,
+				"DuplicatedSetValue",
+				"Variable '%s' appears multiple times in setValues. The operator will keep the last usage with value '%s' (position %d), ignoring the previous value '%s'. This will fail in a future release. Please remove duplicates from setValues.",
```

Comment:
nit: I would clarify the `This` in second sentence to be something like:

Specifying a variable multiple times using setValues will fail in a future release. Please remove duplicates from setValues as it introduces ambiguity.